### PR TITLE
Use kstatus for reconciler deployment status

### PR DESCRIPTION
**Pull request by karlkfi**
- This reduces duplicate code in CS and makes the messages more
  consistent.

### CODE REVIEW

File: pkg/reconcilermanager/controllers/deployment_conditions.go [hunk 1]
```diff
@@ -1,118 +1,29 @@
-// Copyright 2022 Google LLC
```
Comment (sdowell) [inline pkg/reconcilermanager/controllers/deployment_conditions.go:1]: Keep license header

Comment (karlkfi) [reply pkg/reconcilermanager/controllers/deployment_conditions.go]: fixed

**Pull request merged**